src/MongoDB.Driver.Core/MongoWriteConcernException.cs

```diff
@@ -43,9 +43,21 @@ private static void AddErrorLabelsFromWriteConcernResult(MongoWriteConcernExcept
                 }
             }
         }
+
+        private static Exception CreateNestedExceptionOrNull(ConnectionId connectionId, WriteConcernResult writeConcernResult)
```

Comment:
For some reasons we consider writeConcernException only like `document`-like data, but we also can try to map this data into a typed exception from the driver. Needs to be tested.